x/erc20/keeper/ibc_callbacks.go

```diff
@@ -0,0 +1,91 @@
+package keeper
+
+import (
+	sdkerrors "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+
+	channeltypes "github.com/cosmos/ibc-go/v5/modules/core/04-channel/types"
+	"github.com/cosmos/ibc-go/v5/modules/core/exported"
+
+	"github.com/evmos/evmos/v9/ibc"
+	"github.com/evmos/evmos/v9/x/erc20/types"
+	recoverytypes "github.com/evmos/evmos/v9/x/recovery/types"
+)
+
+// OnRecvPacket has been modified to use ConvertCoin.
+// ConvertCoin converts IBC Coins to their ERC-20 Representations,
+// given that the Token Pair is registered through governance.
+// This conversion happens after the original IBC transfer.
+func (k Keeper) OnRecvPacket(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	ack exported.Acknowledgement,
+) exported.Acknowledgement {
+
+	// Setup for OnRecvPacket:
+	// - Get erc20 parameters for keeper
+	// - Get sender/recipient addresses of transfer in `evmos1` and the original bech32 format
+	// - Get denomination of packet's IBC transfer
+	erc20Params := k.GetParams(ctx)
+	sender, recipient, senderBech32, recipientBech32, err := ibc.GetTransferSenderRecipient(packet)
+	if err != nil {
+		return channeltypes.NewErrorAcknowledgement(err)
+	}
+	denom, err := ibc.GetTransferDenomination(packet)
+	if err != nil {
+		return channeltypes.NewErrorAcknowledgement(err)
+	}
+
+	// Check and return error ACK if:
+	// - ERC20s are disabled
+	// - Sender or recipient addresses are blocked
+	if !(erc20Params.GetEnableErc20()) {
+		return channeltypes.NewErrorAcknowledgement(
+			sdkerrors.Wrapf(
+				types.ErrERC20Disabled,
+				"erc20 module is currently disabled by governance",
+			),
+		)
```

Comment:
this shouldn't return error, otherwise all the IBC transfers would fail if the erc20 module is disabled